### PR TITLE
[#492] Get rid of blocking API

### LINKED ISSUE #492: Get rid of blocking API
The blocking API is mainly used for testing but there are some instances of use even outside tests. I believe we can reduce most of the blocking API if not all. Full removal can be achieved by:

1. Converting all tests to async ones which allows us to use non-blocking API instead.
2. Wrap closure based methods to Task.

Blocking API:

```swift
/// Blocking API - used for the testing purposes
public protocol LightWalletServiceBlockingAPI {
    /// Returns the info for this lightwalletd server (blocking)
    func getInfo() throws -> LightWalletdInfo

    ///
    /// Return the latest block height known to the service.
    /// - Parameter result: a result containing the height or an Error
    //func latestBlockHeight() throws -> BlockHeight

    /// Return the given range of blocks.
    ///
    /// - Parameter range: the inclusive range to fetch.
    ///     For instance if 1..5 is given, then every block in that will be fetched, including 1 and 5.
    func blockRange(_ range: CompactBlockRange) throws -> [ZcashCompactBlock]
    
    
    /// Submits a raw transaction over lightwalletd. Blocking
    /// - Parameter spendTransaction: data representing the transaction to be sent
    /// - Throws: LightWalletServiceError
    /// - Returns: LightWalletServiceResponse
    func submit(spendTransaction: Data) throws -> LightWalletServiceResponse
    
    /// Gets a transaction by id
    /// - Parameter txId: data representing the transaction ID
    /// - Throws: LightWalletServiceError
    /// - Returns: LightWalletServiceResponse
    func fetchTransaction(txId: Data) throws -> TransactionEntity
    
    func fetchUTXOs(
        for tAddress: String,
        height: BlockHeight
    ) throws -> [UnspentTransactionOutputEntity]
    
    func fetchUTXOs(
        for tAddresses: [String],
        height: BlockHeight
    ) throws -> [UnspentTransactionOutputEntity]
}
```

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #492 

- blocking API removed, only latestBlockHeight() stayed
- non-blocking closure based API removed
- unit tests updated to use async API

Screenshot 2022-09-30 at 15 26 59

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: Sources/ZcashLightClientKit/Service/LightWalletService.swift [hunk 1]
```diff
@@ -101,117 +101,44 @@ public protocol LightWalletServiceResponse {
...
     /// Return the given range of blocks.
     /// - Parameter range: the inclusive range to fetch.
     ///     For instance if 1..5 is given, then every block in that will be fetched, including 1 and 5.
-    @available(*, deprecated, message: "This function will be removed soon. Use the `blockRange(...) -> AsyncThrowingStream<ZcashCompactBlock, Error>` instead.")
-    func blockRange(_ range: CompactBlockRange, result: @escaping (Result<[ZcashCompactBlock], LightWalletServiceError>) -> Void)
     func blockRange(_ range: CompactBlockRange) -> AsyncThrowingStream<ZcashCompactBlock, Error>
     
     /// Submits a raw transaction over lightwalletd. Non-Blocking
     /// - Parameter spendTransaction: data representing the transaction to be sent
     /// - Parameter result: escaping closure that takes a result containing either LightWalletServiceResponse or LightWalletServiceError
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Service/LightWalletService.swift:124]:
```suggestion
 
```

File: Sources/ZcashLightClientKit/Service/LightWalletService.swift [hunk 2]
```diff
@@ -101,117 +101,44 @@ public protocol LightWalletServiceResponse {
...
-    @available(*, deprecated, message: "This function will be removed soon. Use the `submitAsync(spendTransaction: Data)` instead.")
-    func submit(spendTransaction: Data, result: @escaping(Result<LightWalletServiceResponse, LightWalletServiceError>) -> Void)
-    func submitAsync(spendTransaction: Data) async throws -> LightWalletServiceResponse
+    func submit(spendTransaction: Data) async throws -> LightWalletServiceResponse
     
     /// Gets a transaction by id
     /// - Parameter txId: data representing the transaction ID
     /// - Parameter result: handler for the result
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Service/LightWalletService.swift:129]:
```suggestion
```

File: Tests/TestUtils/FakeService.swift [hunk 3]
```diff
@@ -24,66 +24,38 @@ class MockLightWalletService: LightWalletService {
...
+//    func fetchUTXOs(for tAddresses: [String], height: BlockHeight) throws -> [UnspentTransactionOutputEntity] {
+//        []
+//    }
+//
+//    func fetchUTXOs(
+//        for tAddress: String,
+//        height: BlockHeight,
+//        result: @escaping (Result<[UnspentTransactionOutputEntity], LightWalletServiceError>) -> Void
+//    ) {
+//    }
```
Comment (pacu) [inline Tests/TestUtils/FakeService.swift:47]:
```suggestion
```

File: Tests/TestUtils/FakeService.swift [hunk 4]
```diff
@@ -24,66 +24,38 @@ class MockLightWalletService: LightWalletService {
...
-        height: BlockHeight,
-        result: @escaping (Result<[UnspentTransactionOutputEntity], LightWalletServiceError>) -> Void
-    ) {
-    }
+//    func fetchUTXOs(
+//        for tAddresses: [String],
+//        height: BlockHeight,
+//        result: @escaping (Result<[UnspentTransactionOutputEntity], LightWalletServiceError>) -> Void
+//    ) {
+//    }
```
Comment (pacu) [inline Tests/TestUtils/FakeService.swift:58]:
```suggestion
```

**Review by pacu — CHANGES_REQUESTED** [review]
requesting some changes  on comments and commented code

File: Sources/ZcashLightClientKit/Service/LightWalletService.swift (hunk 1 above)
Comment (LukasKorba) [reply Sources/ZcashLightClientKit/Service/LightWalletService.swift:124]: There were other result parameters, I cleaned up all

File: Tests/TestUtils/FakeService.swift (hunk 3 above)
Comment (LukasKorba) [reply Tests/TestUtils/FakeService.swift:47]: thx for good catch

File: Tests/TestUtils/FakeService.swift (hunk 4 above)
Comment (LukasKorba) [reply Tests/TestUtils/FakeService.swift:58]: thx again

**Review by pacu — APPROVED** [review]

**Pull request merged**